django/contrib/admin/options.py

```diff
@@ -1950,6 +1951,31 @@ def history_view(self, request, object_id, extra_context=None):
             "admin/object_history.html"
         ], context)
 
+    def get_changeformset_initial_data(self, request, prefix_name):
+        """
+        Get the initial formset data from the request's GET params.
+        """
+        initial = dict(request.GET.items())
+        initial_list = []
+        for k in initial:
+            args_list = k.split(sep='_')
+            if re.match(f'{prefix_name}__*_*', k):
```

Comment:
This regex looks wrong. If you're not extracting components of the regex, then why use it when you could just check that the string starts with `prefix_name`?

But given that you are then using `args_list` which is `k` split up it suggests that you do actually want a proper regex with capturing groups.

(Also note that when substituting a string into a regex, `re.escape()` should be used.)